### DISCUSSION TITLE
read_sav from s3 bucket

### DISCUSSION THREAD

**Discussion by NikhilAsalkar** [Roche/pyreadstat#226] [Q&A]
is there any direct / indirect way to read sav file from s3?
as read_sav takes str, bytes or Path-like object to locate the file in a directory.
its unable to read file from s3 unless we download it from a readable location.

referred to an old discussion just wanted to check if we are still on it.

**Comment by jfa501** [comment]
I ran into the same problem for a gcp storage bucket, as far as I know you have to download the object first before reading the file.

**Comment by ofajardo** [comment]
@jfa501 is correct, the underlying C library (Readstat) absolutely needs a file on disk, so you have to copy the file to local. See #138. There is a proposed solution in #8 so it may come some day.
